src/components/PeoplePicker/PeoplePicker.tsx

```diff
@@ -0,0 +1,236 @@
+            value: props.value || props.defaultValue || ''
+        };
+    }
+
+    @autobind
+    private _onInputChange(event: React.ChangeEvent<any>): void {
+        const element: HTMLTextAreaElement = event.target as HTMLTextAreaElement;
+        const value: any = element.value;
+
+        this.setState({
+            value: value
+        });
+
+        if (value.length >= 3) {
+            this.setState({ suggestionsVisible: true });
+            this.props.onSearch(value);
+        } else {
+            this.setState({ suggestionsVisible: false });
+        }
+    }
+
+    private _renderSuggestions(): JSX.Element {
+        return (
+            <div className="people-picker-suggestions">
+                {this.props.loadingSuggestionList && <Spinner type={SpinnerType.small} />}
+                {!this.props.loadingSuggestionList && this.props.suggestionList.map((principal, index) => (
+                    <Principal
+                        principal={principal}
+                        isSelected={false}
+                        onSelect={this._onSuggestionClick}
+                    />
+                ))}
+            </div>
+        );
+    }
+
+    @autobind
+    private _onSuggestionClick(principal: IPrincipal) {
+        this.setState({ suggestionsVisible: false, value: '' });
+
+        if (this.state.selectedPrincipalList !== null) {
+            this.setState({ selectedPrincipalList: [...this.state.selectedPrincipalList, principal] });
+        } else {
+            this.setState({ selectedPrincipalList: [principal] });
+        }
+    }
+
+    @autobind
+    private _renderEmptyField(): JSX.Element {
+        const peoplePickerInputClassName = classNames(
+            'people-picker-input',
+            {
+                'is-disabled': this.props.disabled
+            },
+            [this.props.className]
+        );
+
+        if (!this.props.errorMessage) {
+            return (
+                <input
+                    type={'text'}
+                    ref={(c): HTMLInputElement => this._field = c}
+                    value={this.state.value}
+                    onBlur={this._onBlur}
+                    onFocus={this._onFocus}
+                    onChange={this._onInputChange}
+                    className={peoplePickerInputClassName}
+                    placeholder={this.props.placeholder}
+                />
+            );
+        }
+
+        return (
+            <div className="people-picker-input-error-container">
+                <Tooltip
+                    content={this.props.errorMessage}
+                    className="tooltip-error"
+                    showTooltip={this.state.isFocused}
+                    directionalHint={DirectionalHint.bottomLeftEdge}>
+                    <div className="people-picker-input-error-content">
+                        <input
+                            type={'text'}
+                            ref={(c): HTMLInputElement => this._field = c}
```

Comment:
This should be autobinded function instead of this arrow function. Arrow functions cause unnecessary renders.

Reply:
Ok. It's changed.